required/latex-lab/latex-lab-toc-kernel-changes.dtx

```diff
@@ -187,23 +192,29 @@
 % \subsection{\cs{contentsline} extension}
 % [latex-lab]
 % The tagging code must add code at the begin and end of \cs{contentsline}.
-% This code must have access to the arguments. We set it up as config point
-% for now.
+% This code must have access to the arguments. They are passed
+% together in tagging socket.
+% \begin{socketdecl}{tagsupport/toc/contentsline/before,
+%  tagsupport/toc/contentsline/after}
+%  Tagging sockets at the begin and end of contentsline.
+%  They receive \emph{all} contentsline arguments as one argument
+%  in four brace groups. The socket code should then use the parts it needs.
 %    \begin{macrocode}
-\def\@contentsline@cfgpoint@before#1#2#3#4{}
-\def\@contentsline@cfgpoint@after#1#2#3#4{}
+\NewSocket{tagsupport/toc/contentsline/before}{1}
+\NewSocket{tagsupport/toc/contentsline/after}{1}
 %    \end{macrocode}
+% \end{socketdecl}
 % for the hooks we need that the \cs{l@XX} pass also the destination and
 % that \cs{l@XX} takes three arguments. 
 % TODO: test that, if there are two many problems we should pass the 
 % destination through a command name ...
 %    \begin{macrocode}
 \def\contentsline#1#2#3#4%
   {%
-    \@contentsline@cfgpoint@before{#1}{#2}{#3}{#4}%
+    \UseTaggingSocket{toc/contentsline/before}{{#1}{#2}{#3}{#4}}%
     \gdef\@contentsline@destination{#4}%
     \csname l@#1\endcsname{#2}{#3}%
-    \@contentsline@cfgpoint@after{#1}{#2}{#3}{#4}%
+    \UseTaggingSocket{toc/contentsline/after}{{#1}{#2}{#3}{#4}}%
```

Comment:
one further comment here: contentslines do not need to be vertical, e.g., they may start out vertical but stay in horizontal mode (e.g., some of the titletoc stuff) so I wonder if there should be a before/after or perhaps stuff added to the definitions of `\l@...`.   That might complicate things, but the before after may not work in such cases.

Reply:
the code of these sockets is imho mode independant. The h/v-mode question is elsewhere. But reviewing that is for another day, when one tries to really extend the toc support to titletoc etc. For now I only wanted to get rid of the configuration points.

Reply:
ok